src/lfortran/semantics/ast_body_visitor.cpp

```diff
@@ -3428,6 +3428,10 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             } else if ( locality->type == AST::concurrent_localityType::ConcurrentLocal ) {
                 AST::ConcurrentLocal_t *private_ = AST::down_cast<AST::ConcurrentLocal_t>(locality);
                 for ( size_t j = 0; j < private_->n_vars; j++ ) {
+                    // check if loop variable is part of local expr
+                    if (current_scope->resolve_symbol(to_lower(private_->m_vars[j])) == ASR::down_cast<ASR::Var_t>(var)->m_v ) {
+                        throw SemanticError("Loop variable `" + std::string(private_->m_vars[j]) + "` cannot be part of local expression", private_->base.base.loc);
```

Comment:
Maybe like this:
```suggestion
                        throw SemanticError("Do concurrent loop variable `" + std::string(private_->m_vars[j]) + "` cannot be part of local expression", private_->base.base.loc);
```
since other loop variables can be part of it, just not do concurrent one.